### PR TITLE
Fixes #220 sslproto memory issue

### LINKED ISSUE #220: memory leak with 0.12.0
* **uvloop version**: 0.12.0
* **Python version**: 3.7.2
* **Platform**: Mac os and Docker
* **Can you reproduce the bug with `PYTHONASYNCIODEBUG` in env?**: Have not tried yet

Recently I upgraded from `0.11.3` to `0.12.0`. I use uvloop to constantly pull data from an API and insert into a database.  After running for a couple of minutes, I found that my memory usage jumped from ~100MB to ~450mb.  Has anybody else run into this issue? I have a lot going on so reproducing it with a smaller gist would be hard right now. But reverting back to `0.11.3` immediately fixed the issue.  So I am guessing something is related to the upgrade.

### PR DESCRIPTION
**Pull request by fantix**
I think this could at least fix the issue that is reproducible with the given script in #220. In the meantime, I'm still looking for more corner cases that may lead to issues.

- [x] Write tests to cover #220 

Tasks below shall be addressed in a new PR:

- [ ] `SSLTransport` that is never closed by orphaned AppProtocol (shared with asyncio?)
- [ ] uvloop should clear reference to the callback (not just args, new PR)
- [ ] Review all corner cases and write more tests

### CODE REVIEW

File: uvloop/sslproto.pyx [hunk 1]
```diff
@@ -469,7 +469,7 @@ cdef class SSLProtocol:
         else:
             self._do_handshake()
 
-    cdef _check_handshake_timeout(self):
+    def _check_handshake_timeout(self):
```
[the remark this answers is not in the document]
Comment (fantix) [reply uvloop/sslproto.pyx]: Oh sure, about to amend that commit with tests and possible further fixes too. Do you think we should merge this initial fix first as a hotfix?

[the remark this answers is not in the document]
Comment (1st1) [reply uvloop/sslproto.pyx]: I'm OK to wait for the amended commit a bit.  We can release 0.12.1 tomorrow then, if the memory leak issue is fixed!

[the remark this answers is not in the document]
Comment (fantix) [reply uvloop/sslproto.pyx]: I see, let me try to split the work with more details, and let you know when the first one is ready for review and merge.

**Comment by fantix** [comment]
OK if tests pass I think this PR is good enough now, please kindly review.

**Comment by 1st1** [comment]
> OK if tests pass I think this PR is good enough now, please kindly review.
Looks like it's red :(

**Comment by fantix** [comment]
Oh I'm stupid, has been investigating why `Test_UV_TCPSSL.test_handshake_timeout_handler_leak` passes locally while Travis is red ... it is actually failing for `Test_AIO_TCPSSL`!! Let me see

**Comment by fantix** [comment]
Yay! Green now. Also added a second test.
